### PR TITLE
Update Migration Guide

### PR DESCRIPTION
**Pull request by uhbrar**
This adds a guide dedicated more to KP's as opposed to ARA's. This also includes an example for source retrieval provenance refactoring.

### CODE REVIEW

**Comment by edeutsch** [comment]
You can more easily read the new formatted guide here: https://github.com/uhbrar/ReasonerAPI/blob/update_guide/MigrationAndImplementationGuide1-4.md

**Review by edeutsch — COMMENTED** [review]
There seemed to be some substantial push-back against implementing this policy when we brought this up, so I think this will need some more discussion before merging.

File: MigrationAndImplementationGuide1-4.md
Comment (edeutsch) [inline MigrationAndImplementationGuide1-4.md:1048]: I don't think results have edge_bindings any more?

**Review by edeutsch — CHANGES_REQUESTED** [review]

File: MigrationAndImplementationGuide1-4.md
Comment (uhbrar) [reply MigrationAndImplementationGuide1-4.md]: Resolved

**Comment by uhbrar** [comment]
I thought that it was pretty well accepted that subclasses could be represented this way when discussed at architecture. There was definitely discussion about this use case being added to the guide.

**Comment by edeutsch** [comment]
My recollection from the TRAPI call is that this was discussed and there was substantial push-back, perhaps from Vlado and others, I don't recall now exactly. The notes say:
```
There was mention of KPs potentially using AuxiliaryGraph for subclass encoding? Do we want to pursue that? Come up with a recommendation/example on how to encode a subclass reasoning response from a KP? Point people: Chris, Abrar, Matt, Andrew?
For now, the policy is to just keep doing it the current way, and we will consider a change in the future
[Eric] will start a thread to discuss if it is sensible to create a subclass example. DONE
```

I started the issue here: https://github.com/NCATSTranslator/ReasonerAPI/issues/414
There is some discussion, but I suspect others will want to contribute.

I think this change will need formal approvals before merging.
There are multiple revisions in this PR so it is hard to see the specific proposed changed, but this link may be more helpful:
https://github.com/NCATSTranslator/ReasonerAPI/pull/420/commits/e9c41d59a8e9e30f28f9491b4bbb970b80bae8d6

I suggest conversion continue in the issue.
And the topic could be revisited in the Architecture and/or TRAPI calls next week.

File: MigrationAndImplementationGuide1-4.md
Comment (colleenXu) [inline MigrationAndImplementationGuide1-4.md:1]: focusses -> focuses
Also apostrophe use? For plural, probably use "ARAs", "KPs", and "CURIEs".

File: MigrationAndImplementationGuide1-4.md
Comment (colleenXu) [inline MigrationAndImplementationGuide1-4.md:337]: is this supposed to be "biolink:support_graphs", from [here](https://github.com/biolink/biolink-model/blob/35925815e6ef47c0e4a0e28417452be486c1c18f/biolink-model.yaml#L672)?

File: MigrationAndImplementationGuide1-4.md
Comment (colleenXu) [inline MigrationAndImplementationGuide1-4.md:572]: see https://github.com/NCATSTranslator/ReasonerAPI/pull/420/files#r1162416075

File: MigrationAndImplementationGuide1-4.md
Comment (colleenXu) [inline MigrationAndImplementationGuide1-4.md:978]: should use MONDO:0005015 (plain diabetes m.). This curie is actually for type2...

File: MigrationAndImplementationGuide1-4.md
Comment (colleenXu) [reply MigrationAndImplementationGuide1-4.md]: I recall in the Architecture call 2 weeks ago, there was a question on whether KPs could use result.edge_bindings (particularly since they are often called to do one-hop edge-retrieval by ARAs). Was this question resolved? Does it depend on the subclassing discussion?

**Comment by colleenXu** [comment]
Also does the description for Result need editing since it discusses edges and edge-bindings? https://github.com/uhbrar/ReasonerAPI/blob/1.4/TranslatorReasonerAPI.yaml#L564

File: MigrationAndImplementationGuide1-4.md
Comment (uhbrar) [reply MigrationAndImplementationGuide1-4.md]: Fixed. Thank you.

There's noting in the specification that necessarily says that it has to be a list, but I think expected practice is a list as opposed to a single value. Unless this was further specified in the biolink model.

File: MigrationAndImplementationGuide1-4.md
Comment (uhbrar) [reply MigrationAndImplementationGuide1-4.md]: Fixed.

File: MigrationAndImplementationGuide1-4.md
Comment (uhbrar) [reply MigrationAndImplementationGuide1-4.md]: Fixed. Thank you.

**Comment by uhbrar** [comment]
I didn't notice that. I thought I'd already fixed the description for Result. I'll put in a separate pull request in for that.

File: MigrationAndImplementationGuide1-4.md
Comment (colleenXu) [inline MigrationAndImplementationGuide1-4.md:1]: still has a "focusses" typo

**Review by edeutsch — APPROVED** [review]

**Review by vdancik — APPROVED** [review]

**Review by edeutsch — APPROVED** [review]

**Review by edgargaticaCU — APPROVED** [review]

**Review by RichardBruskiewich — APPROVED** [review]

**Comment by edeutsch** [comment]
The number of approvals is very meagre, but I'm going to merge this anyway, because if we actually wait for approvals we'll never merge it.

**Pull request merged**
